db/compaction/compaction_job.cc

```diff
@@ -781,7 +781,7 @@ Status CompactionJob::Run() {
             cfd->internal_comparator(), files_output[file_idx]->meta,
             /*range_del_agg=*/nullptr,
             compact_->compaction->mutable_cf_options(),
-            /*table_reader_ptr=*/nullptr,
```

Comment:
This is the part that we care about the most IMO. We want to have `internal_stats` passed in from `CompactionJob::Run`